src/shared/utilities/timeoutUtils.ts

```diff
@@ -18,6 +19,19 @@ export class TimeoutError extends Error {
     public static isCancelled(err: any): err is TimeoutError & { type: 'cancelled' } {
         return err instanceof TimeoutError && err.type === 'cancelled'
     }
+
+    public static isExpired(err: any): err is TimeoutError & { type: 'expired' } {
+        return err instanceof TimeoutError && err.type === 'expired'
+    }
+}
+
+interface CancelEvent {
+    readonly type: 'expired' | 'cancelled'
+}
```

Comment:
Is this necessary? Would it just be better to do something like `type CancelEvent = 'expired' | 'cancelled'`? It's a bit strange seeing the references to `CancelEvent['type']` when it's literally just the strings.

Reply:
Not needed, it's probably just me making interfaces habitually. Having the interface would make it easier to extend, assuming it was ever actually extended :)

Beyond that it's just noise.